seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/SinkFlowLifeCycle.java

```diff
@@ -246,8 +271,43 @@ public void received(Record<?> record) {
                 if (prepareClose) {
                     return;
                 }
+                AtomicReference<String> tableId = new AtomicReference<>();
                 writer.write((T) record.getData());
-                taskMetricsCalcContext.updateMetrics(record.getData());
+                if (record.getData() instanceof SeaTunnelRow) {
+                    if (this.sinkAction.getSink() instanceof MultiTableSink) {
+                        if (((SeaTunnelRow) record.getData()).getTableId() == null
+                                || ((SeaTunnelRow) record.getData()).getTableId().isEmpty()) {
+                            tableId.set(((SeaTunnelRow) record.getData()).getTableId());
+                        } else {
+                            tablesMaps.forEach(
+                                    tablePathTablePathMap -> {
+                                        tablePathTablePathMap.forEach(
+                                                (k, v) -> {
+                                                    if (k.equals(
+                                                            TablePath.of(
+                                                                    ((SeaTunnelRow)
+                                                                                    record
+                                                                                            .getData())
+                                                                            .getTableId()))) {
+                                                        tableId.set(v.getFullName());
+                                                    }
+                                                });
+                                    });
+                        }
```

Comment:
I think using foreach to filter will have performance issues, especially when there are many tables, because it will be executed once for each row.